### PR TITLE
Stream 827/create poc quantum models

### PR DESCRIPTION
**Pull request by ShahNewazKhan**
- Adds POC `quantum` model `sportsdb`  
- Disables dbt docs GHA
- Adds `SF GHA Task` artifacts
- Updates `xyz-models` to `datascience-models` documentation
- Sets up `dbt_project.yml` for `livequery` models & `API_INTEGRATION/EXTERNAL_URL` vars
- Adds `fsc_utils == 1.21.7`
- Adds `api_integration` macro

[Cloud Watch logs](https://us-east-1.console.aws.amazon.com/cloudwatch/home?region=us-east-1#logsV2:log-groups/log-group/$252Faws$252Flambda$252Fdatascience-api-stg-WorkerBulkRestApi-6gNo3EiKPIsl/log-events/2024$252F04$252F05$252F$255B$2524LATEST$255D3debc4fb50a94e7b8279c209d59469e0) of `quantum` model invocation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,6 +26,61 @@ datascience:
...
+
+## Invoking the POC Quantum model
+
+You will find a `POC` `quantum` model in [models/streamline/quantum/streamline__sportsdb_live_scorers.sql](/models/streamline/quantum/streamline__sportsdb_live_scorers.sql).  
+
+This model is intended to be used as a `POC` for showcasing the capabilities of using `Quantum Models` with a `streamine` backend wherein a single `DBT` model can be used for both `pull` and `push` based workloads. 
+
+The is a `makefile` [directive](./Makefile#L2) for invoking the `POC` quantum model. To invoke the model, run the following make command `make quantum-poc`. Invoking `make quantum-poc` will run the `POC` quantum model and the output will be as follows :
```
Comment (ShahNewazKhan) [inline README.md:39]:
```suggestion
There is a `makefile` [directive](./Makefile#L2) for invoking the `POC` quantum model. To invoke the model, run the following make command `make quantum-poc`. Invoking `make quantum-poc` will run the `POC` quantum model and the output will be as follows :
```

File: README.md [hunk 2]
```diff
@@ -26,6 +26,61 @@ datascience:
...
+03:45:50  2 of 2 START hook: livequery_models.on-run-end.0 ............................... [RUN]
+03:45:50  2 of 2 OK hook: livequery_models.on-run-end.0 .................................. [OK in 0.00s]
+03:45:50  
+03:45:50  
+03:45:50  Finished running 1 view model, 5 hooks in 0 hours 0 minutes and 13.73 seconds (13.73s).
+03:45:50  
+03:45:50  Completed successfully
+03:45:50  
+03:45:50  Done. PASS=1 WARN=0 ERROR=0 SKIP=0 TOTAL=1
+```
```
Comment (ShahNewazKhan) [inline README.md:83]: **Note:** For more details on using the `udf params` in the models post hook refer to this [discussion](https://github.com/FlipsideCrypto/streamline-flow/discussions/10#discussioncomment-7194378).

Comment (ShahNewazKhan) [inline README.md:83]:
```suggestion
```
**Note:** For more details on using the `udf params` in the models post hook refer to this [discussion](https://github.com/FlipsideCrypto/streamline-flow/discussions/10#discussioncomment-7194378).

```

File: README.md [hunk 3]
```diff
@@ -26,6 +26,61 @@ datascience:
...
 ### Resources:
```
Comment (ShahNewazKhan) [inline README.md:84]:
```suggestion
**Note:** For more details on using the `udf params` in the models post hook refer to this [discussion](https://github.com/FlipsideCrypto/streamline-flow/discussions/10#discussioncomment-7194378).

### Resources:
```

File: README.md [hunk 4]
```diff
@@ -26,6 +26,66 @@ datascience:
       query_tag: <TAG>
 ```
 
+### Project setup
+
+This project has been setup with `fsc_utils == 1.21.7` according to the instructions in the  [fsc_utils setup](https://github.com/FlipsideCrypto/fsc-utils?tab=readme-ov-file#adding-the-fsc_utils-dbt-package) documentation. In addition `livequery` & `livequery marketplace GHA` models, schemas & functions has also been deployed to the `DATASCIENCE_DEV` database as per the instructions in the [livequery setup](https://github.com/FlipsideCrypto/fsc-utils?tab=readme-ov-file#livequery-functions) & [snowflake GHA tasks setup](https://github.com/flipsideCrypto/fsc-utils?tab=readme-ov-file#snowflake-tasks-for-github-actions) documentations.
```
Comment (ShahNewazKhan) [inline README.md:31]:
```suggestion
This project has been setup with `fsc_utils == 1.21.7` according to the instructions in the [fsc_utils setup](https://github.com/FlipsideCrypto/fsc-utils?tab=readme-ov-file#adding-the-fsc_utils-dbt-package) documentation. In addition `livequery` & `livequery marketplace GHA` models, schemas & functions has also been deployed to the `DATASCIENCE_DEV` database as per the instructions in the [livequery setup](https://github.com/FlipsideCrypto/fsc-utils?tab=readme-ov-file#livequery-functions) & [snowflake GHA tasks setup](https://github.com/flipsideCrypto/fsc-utils?tab=readme-ov-file#snowflake-tasks-for-github-actions) documentations.
```

File: packages.yml [hunk 5]
```diff
@@ -2,4 +2,4 @@ packages:
   - package: calogica/dbt_expectations
     version: [">=0.4.0", "<0.9.0"]
   - git: https://github.com/FlipsideCrypto/fsc-utils.git
-    revision: "v1.20.0"
\ No newline at end of file
+    revision: "v1.21.7-dev"
```
Comment (juls858) [inline packages.yml:5]: should dev be removed?

File: requirements.txt [hunk 6]
```diff
@@ -0,0 +1 @@
+dbt-snowflake==1.7.0
```
Comment (juls858) [inline requirements.txt:1]: we shouldn't pin patch versions

File: .github/workflows/dbt_docs_update.yml [hunk 7]
```diff
@@ -1,9 +1,10 @@
 name: docs_update
 
 on:
-  push:
-    branches:
-      - "main"
+  workflow_dispatch:
+  # push:
```
Comment (juls858) [inline .github/workflows/dbt_docs_update.yml:5]: can we delete these?

**Review by juls858 — CHANGES_REQUESTED** [review]

File: packages.yml (hunk 5 above)
Comment (ShahNewazKhan) [reply packages.yml:5]: good catch, ya need to update to [v1.21.7 release ](https://github.com/FlipsideCrypto/fsc-utils/releases/tag/v1.21.7)

Comment (ShahNewazKhan) [reply packages.yml:5]: updated accordingly

File: requirements.txt (hunk 6 above)
Comment (ShahNewazKhan) [reply requirements.txt:1]: updated accordingly

File: .github/workflows/dbt_docs_update.yml (hunk 7 above)
Comment (ShahNewazKhan) [reply .github/workflows/dbt_docs_update.yml]: deleted

**Review by juls858 — APPROVED** [review]

**Pull request merged**